aten/src/ATen/native/Dropout.cpp

```diff
@@ -117,6 +117,9 @@ Tensor native_dropout_backward_cpu(const Tensor& grad, const Tensor& mask, doubl
 Tensor dropout(const Tensor& input, double p, bool train) {
   auto result = [&]() {
     NoNamesGuard guard;
+    if (input.is_nested()) {
```

Comment:
As discussed offline, I think we should avoid `is_nested()` calls whenever possible, letting the dispatcher do the work. Given that, I'd suggest leaving dropout in its current state of having both `CompositeImplicitAutograd` and `NestedTensor*` keys registered to it (and similarly for linear).